src/js/messaging/reducers/compose.js

```diff
@@ -0,0 +1,15 @@
+import set from 'lodash/fp/set';
+import { SET_CATEGORY } from '../actions/compose.js';
+
+const initialState = {
+  category: undefined
+};
+
+export default function compose(state = initialState, action) {
+  switch (action.type) {
+    case SET_CATEGORY:
+      return set('category', action.value.value, state);
```

Comment:
Maybe making it `action.category.value` would be more readable and less confusing. It often struck me as a minor eyesore when we used `action.data.data` in other places.